catt/cli.py

```diff
@@ -16,29 +17,35 @@ def get_local_ip(cc_host):
     return s.getsockname()[0]
 
 
-class Device(object):
-    def __init__(self, name):
-        self.name = name
-
-
-pass_device = click.make_pass_decorator(Device)
+class CattCliError(click.ClickException):
+    pass
 
 
 @click.group()
-@click.option("--delete-cache", is_flag=True, help="Empty the Chromecast discovery cache.")
-@click.option("-d", "--device", metavar="NAME", help="Select Chromecast device.")
+@click.option("--delete-cache", is_flag=True,
+              help="Empty the Chromecast discovery cache.")
+@click.option("--write-config", is_flag=True,
+              help="Write name of default Chromecast device to config file")
+@click.option("-d", "--device", metavar="NAME",
+              help="Select Chromecast device.")
 @click.pass_context
-def cli(ctx, delete_cache, device):
+def cli(ctx, delete_cache, write_config, device):
     if delete_cache:
         Cache().clear()
-    ctx.obj = Device(device)
+    ctx.obj["device"] = device
+    if write_config:
+        if device:
+            cast = CastController(device)
+            writeconfig()
```

Comment:
I think `writeconfig` should definitely accept the `settings` object here, rather than get it through `click`. I'm also not crazy about it being called `writeconfig` instead of `write_config`, but I can see the collision...